services-api/src/main/java/io/scalecube/services/ServicesProvider.java

```diff
@@ -0,0 +1,34 @@
+package io.scalecube.services;
+
+import java.util.Collection;
+import reactor.core.publisher.Mono;
+
+/** Manages the life cycle of all services registered with Scale Cube Services. */
+public interface ServicesProvider {
+
+  /**
+   * Provide service definitions.
+   *
+   * @param microservices scale cube instance.
+   * @return collection of service definitions - service type&amp;tags.
+   * @see ServiceDefinition
+   */
+  Mono<? extends Collection<ServiceDefinition>> provideServiceDefinitions(
```

Comment:
Remove `Mono` use good-old collection. Name it `getServiceDefinitions()`.

Reply:
Nope. ServiceDefinition maybe blocking operation. Example. ScalecubeServicesProvider can only give out the types of services after their initialization.